engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
How would it make sense to have the rug pulled out from under Thread-1? Perhaps a ThreadLocal makes more sense...